frontend/src/components/visual-editor/v2/Diagrams.tsx

```diff
@@ -299,135 +306,166 @@ const Diagrams = () => {
     ),
   ]);
 
+  const handleLinkDeletion = async (linkId: string) => {
+    const link = model?.getLink(linkId) as any;
+    const sourceId = link?.sourcePort.parent.options.id;
+    const targetId = link?.targetPort.parent.options.id;
+
+    if (link?.sourcePort.options.label === BlockPorts.nextBlocksOutPort) {
+      await removeNextBlockLink(sourceId, targetId);
+    } else if (
+      link?.sourcePort.options.label === BlockPorts.attachmentOutPort
+    ) {
+      await removeAttachmentLink(sourceId, targetId);
+    }
+  };
+  const removeNextBlockLink = async (sourceId: string, targetId: string) => {
+    const previousData = getBlockFromCache(sourceId);
+    const nextBlocks = [...(previousData?.nextBlocks || [])];
+
+    await updateBlock(
+      {
+        id: sourceId,
+        params: {
+          nextBlocks: nextBlocks.filter((block) => block !== targetId),
+        },
+      },
+      {
+        onSuccess() {
+          updateCachedBlock({
+            id: targetId,
+            preprocess: ({ previousBlocks = [], ...rest }) => ({
+              ...rest,
+              previousBlocks: previousBlocks.filter(
+                (block) => block !== sourceId,
+              ),
+            }),
+          });
+        },
+      },
+    );
+  };
+  const removeAttachmentLink = async (sourceId: string, targetId: string) => {
+    await updateBlock(
+      {
+        id: sourceId,
+        params: { attachedBlock: null },
+      },
+      {
+        onSuccess() {
+          updateCachedBlock({
+            id: targetId,
+            preprocess: (oldData) => ({ ...oldData, attachedToBlock: null }),
+          });
+        },
+      },
+    );
+  };
+  const handleBlockDeletion = async (blockIds: string[]) => {
+    if (blockIds.length > 1) {
+      await deleteMultipleBlocks(blockIds);
+    } else {
+      await deleteSingleBlock(blockIds[0]);
+    }
+  };
```

Comment:
handleBlockDeletion could be simplified by directly calling deleteBlocks for both single and multiple block deletions.